### PR TITLE
Add let to const refactor

### LINKED ISSUE #13133: add codefix that rewrites var to let to const where appropriate
consider adding a codefix that transforms `var` to `let` to `const` by picking strictest declaration possible

### PR DESCRIPTION
**Pull request by Kingwl**
Fixes #13133

### CODE REVIEW

File: src/services/refactors/convertLetToConst.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
+/* @internal */
+namespace ts.refactor.generateGetAccessorAndSetAccessor {
```
Comment (AlCalzone) [inline src/services/refactors/convertLetToConst.ts:2]: Should that really be called "generateGetAccessorAndSetAccessor"?

Comment (Kingwl) [reply src/services/refactors/convertLetToConst.ts:2]: Emmm.. Copy & Paste. You know.... :XD

**Comment by jessetrinity** [comment]
While this is a cool capability, this feature seems more appropriate as a lint rule. ESLint already has [prefer-const](https://eslint.org/docs/rules/prefer-const).

I do realize that the associated issue being left open probably means we wanted this at some point, but maybe ESLint didn't do this at the time.

@DanielRosenwasser FYI the issue we discussed offline if you wanted to add anything.

**Comment by Kingwl** [comment]
What about add the new feature about readonly property declaration? I'm not sure could eslint track property assignment by symbol . :XD

**Comment by jessetrinity** [comment]
@Kingwl is there an open suggestion item? Also these kinds of conversions seem like they would be more suited to code fixes if we had an easy way to enable/disable specific diagnostics...

**Pull request closed without merging**